spec/ruby/core/env/delete_spec.rb

```diff
@@ -16,9 +16,25 @@
     ENV.delete("foo").should == "bar"
   end
 
+  it "ignores the block if the environment variable exists" do
+    ENV["foo"] = "bar"
+    begin
+      -> { ENV.delete("foo") { |name| fail name } }.should_not raise_error(RuntimeError)
```

Comment:
I would test the return value as well, or that the variable no longer exist after, `should_not raise_error` doesn't test anything:
```ruby
ENV.delete("foo") { |name| fail name }.should == "bar"
```